RetailCoder.VBE/Refactorings/Rename/RenameRefactoring.cs

```diff
@@ -401,127 +541,60 @@ private void RenameUsages(Declaration target, string interfaceName = null)
-                            var newMemberName = _model.NewName + '_' + member.IdentifierName;
-                            var method = _model.Declarations.Named(oldMemberName).SingleOrDefault(m => m.QualifiedName.QualifiedModuleName == grouping.Key);
-                            if (method == null)
-                            {
-                                continue;
-                            }
-
-                            var content = module.GetLines(method.Selection.StartLine, 1);
-                            var newContent = GetReplacementLine(content, newMemberName, member.Selection);
-                            module.ReplaceLine(method.Selection.StartLine, newContent);
-                        }
-                    }
                 }
             }
         }
 
-        private string GetReplacementLine(string content, string newName, Selection selection)
+        private static void RenameDeclaration(RenameModel model, Declaration target)
         {
-            var contentWithoutOldName = content.Remove(selection.StartColumn - 1, selection.EndColumn - selection.StartColumn);
-            return contentWithoutOldName.Insert(selection.StartColumn - 1, newName);
+            RenameDeclaration(model, target, model.NewName);
         }
 
-        private string GetReplacementLine(ICodeModule module, Declaration target, string newName)
+        private static void RenameDeclaration(RenameModel model, Declaration target, string newName)
         {
-            var content = module.GetLines(target.Selection.StartLine, 1);
-            
-            if (target.DeclarationType == DeclarationType.Parameter)
-            {
-                var rewriter = _model.State.GetRewriter(target.QualifiedName.QualifiedModuleName.Component);
-
-                var identifier = ((VBAParser.ArgContext)target.Context).unrestrictedIdentifier();
-                rewriter.Replace(identifier, _model.NewName);
-
-                // Target.Context is an ArgContext, its parent is an ArgsListContext;
-                // the ArgsListContext's parent is the procedure context and it includes the body.
-                var context = (ParserRuleContext)target.Context.Parent.Parent;
-                var firstTokenIndex = context.Start.TokenIndex;
-                var lastTokenIndex = -1; // will blow up if this code runs for any context other than below
-
-                var subStmtContext = context as VBAParser.SubStmtContext;
-                if (subStmtContext != null)
-                {
-                    lastTokenIndex = subStmtContext.argList().RPAREN().Symbol.TokenIndex;
-                }
 
-                var functionStmtContext = context as VBAParser.FunctionStmtContext;
-                if (functionStmtContext != null)
+            var component = target.QualifiedName.QualifiedModuleName.Component;
+            var rewriter = model.GetRewriter(target);
+            var module = component.CodeModule;
+            {
```

Comment:
Probably a leftover from when I first implemented the COM wrappers; the early implementation disposed every single COM object as soon as it was used. There are probably a couple of extraneous scopes sprinkled left and right in the code base - that would be one of them. Kill on sight.